platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java

```diff
@@ -59,22 +60,33 @@ public int compare(V first, V second) {
 
 			if (isArgumentsCastableTo(first, second, IDisplayable.class)) {
 				var comparator = new CollatingDisplayStringFunctionBasedComparator<>(displayStringFunctionSupplier.get());
-				result = Comparator.nullsFirst(comparator).compare(first, second);
+				result = compareNullsFirst(comparator, first, second);
 			}
 
 			if (result == 0) {
 				if (isArgumentsCastableTo(first, second, IBasicItem.class)) {
 					var comparator = BasicItemComparator.get();
-					result = Comparator.nullsFirst(comparator).compare((IBasicItem) first, (IBasicItem) second);
+					result = compareNullsFirst(comparator, (IBasicItem) first, (IBasicItem) second);
 				}
 			}
 
 			return result;
 		}
 
 		else {
+			if (isArgumentsCastableTo(first, second, Integer.class)) {
```

Comment:
for everything but BigDecimal, you could use Number.doubleValue
```
if (isArgumentsCastableTo(first, second, Number.class)) {
    return compareNullsFirst(Double.compareTo, toNumberDoubleValue(first), toNumberDoubleValue(second));
}
```
and event for BigDecimal, i think Number.doubleValue might suffice, if not then do the BigDecimal check before the Number check.